spec/controllers/openid_connect/authorization_controller_spec.rb

```diff
@@ -57,7 +57,8 @@
                  unauthorized_scope: true,
                  user_fully_authenticated: true,
                  acr_values: 'http://idmanagement.gov/ns/assurance/ial/1',
-                 scope: 'openid')
+                 scope: 'openid',
+                 code_digest: anything)
```

Comment:
in an ideal world, this should be `kind_of(String)` however, due to the mess of redirects, side effects, what-not, this ends up being `nil` in these controller specs, which is super unfortunate.

I would really like to reorganize the interactions between:
- The OIDC::AuthorizationController
- The `IdentityLinker` class
- and the `OpenIdConnectAuthorizeForm`

since there are some side effecty, ordering, inter-dependencies that I am not a fan of right now

Reply:
🤠 😭 😎 turns out this is nil in real life too... time for some gross code changes

Reply:
I'm not proud of this but 33bf61e4bbd131b4c3e051d87bbd6f535e299d34 seems to cover this and I tried to add explanatory comments